Thesis/FunBigStepSILR2.agda

```diff
@@ -325,12 +334,18 @@ mutual
     Σ[ dv ∈ Val (Δτ τ) ] Σ[ dn ∈ ℕ ] eval dt dρ dn ≡ Done dv 0 ×
     relV3 τ v1 dv v2 (suc n-j)
```

Comment:
There's still one mistake here in the definition of related computations: for incremental computation we can't prove that `t2` terminates in general. To get a counterexample to the fundamental lemma there, consider program `if x == 0 then 1 else loop` with inputs `x1 = 0` and `x2 = 1`.

@yurug's definition is better here. But would it go through? Luckily we know from "Imperative self-adjusting computation": there, `t2 terminates in an unspecified number of steps` is indeed an assumption.

But the conclusion is *still* be that outputs v1, dv and v2 are related for `k - j` steps.

That is, to sum up: computations t1, dt and t2 are related at k steps, if, when t1 terminates in j < k steps to v1 and t2 terminates to v2, then dt terminates to dv, and v1, dv and v2 are related values for k - j steps.

And as discussed, all this needs step counts equivalent to small-step reduction, not any other metric.

Also, we do need the mutually inductive logical relations, for computations and values. We can't merge them.